tests/geopage/models.py

```diff
@@ -161,70 +157,145 @@ class GeoPageWithLeaflet(Page):
+                            ),
+                        ],
+                        icon="user",
+                    ),
                 ),
-            ),
-        ]
-    )
+            ],
+            use_json_field=True,
+        )
+    else:
+        body = StreamField(
+            [
+                ("map", GoogleMapsBlock()),
+                ("map_with_leaflet", LeafletBlock()),
+                (
+                    "map_struct",
+                    blocks.StructBlock(
+                        [
+                            ("address", GeoAddressBlock(required=True)),
+                            ("map", GoogleMapsBlock(address_field="address")),
+                        ],
+                        icon="user",
+                    ),
+                ),
+                (
+                    "map_struct_with_deprecated_geopanel",
+                    blocks.StructBlock(
+                        [
+                            ("address", blocks.CharBlock(required=True)),
+                            ("map", GeoBlock(address_field="address")),
+                        ],
+                        icon="user",
+                    ),
+                ),
+                (
+                    "map_struct_with_leaflet",
+                    blocks.StructBlock(
+                        [
+                            ("address", GeoAddressBlock(required=True)),
+                            ("map", LeafletBlock(address_field="address")),
+                        ],
+                        icon="user",
+                    ),
+                ),
+                (
+                    "map_struct_with_zoom",
+                    blocks.StructBlock(
+                        [
+                            ("address", GeoAddressBlock(required=True)),
+                            ("zoom", GeoZoomBlock(required=False)),
+                            (
+                                "map",
+                                GoogleMapsBlock(
+                                    address_field="address", zoom_field="zoom"
+                                ),
+                            ),
+                        ],
+                        icon="user",
+                    ),
+                ),
+                (
+                    "map_struct_leaflet_with_zoom",
+                    blocks.StructBlock(
+                        [
+                            ("address", GeoAddressBlock(required=True)),
+                            ("zoom", GeoZoomBlock(required=False)),
+                            (
+                                "map",
+                                LeafletBlock(
+                                    address_field="address", zoom_field="zoom"
+                                ),
+                            ),
+                        ],
+                        icon="user",
+                    ),
+                ),
+            ]
+        )
 
     content_panels = Page.content_panels + [
         StreamFieldPanel("body"),
```

Comment:
This can be `FieldPanel("body"),` where wagtail is at v3

Reply:
I removed the import alias for wagtail 3 and placed the content_panel in a conditional.